pgpubsub/models.py

```diff
@@ -20,13 +22,32 @@ class Notification(models.Model):
     # nullable as in it always gets a value.
     # After some time the field should be made non nullable here.
     created_at = models.DateTimeField(auto_now_add=True, null=True)
+    db_version = models.IntegerField(null=True)
+
+    class Meta:
+        triggers = [
+            pgtrigger.Trigger(
+                name="pgpubsub_notification_set_db_version",
+                operation=pgtrigger.Insert | pgtrigger.Update,
+                when=pgtrigger.Before,
+                func="""
+                    NEW.db_version := (
+                        SELECT max(id)
+                        FROM django_migrations
+                        WHERE app = ((NEW.payload #>> '{}')::jsonb ->> 'app')
```

Comment:
While working on this I've found that `payload` field doesn't actually contain a JSON with fields `app`, `model` etc. It contains a string value which is a serialized json. 

I've tried to change that but this seems to be done intentionally. I understand that a string is used to be able to find and lock the `Notification` entry by the payload when the notification is received via `LISTEN` in the the `TriggerChannel`. I guess a correct way to fix this is to not search by the whole payload, but instead parse the payload and take some idendifier from it (it can be PK of the Notification) .

Reply:
Hmm, I don't think I did do that intentionally. I thought the payload was being serialized as JSON via the `to_json` call in the trigger:
```
class LockableNotify(Notify):
    def _pre_notify(self):
        return f'''
            INSERT INTO pgpubsub_notification (channel, payload, created_at)
            VALUES ('{self.name}', to_json(payload::text), NOW());
        '''
``` 

Any idea why that wouldn't work? Raw postgres column manipulation and the like isn't my strong point, so it's probably something silly I've done.

I don't think, however, that the payload needs to be a string for it to be used as a search term in the Notification query, e.g.
```
Notification.objects.select_for_update(
                skip_locked=True).filter(
                channel=self.notification.channel,
                payload=self.notification.payload,
            ).first()
```

Doesn't Django allow you to search directly with JSON on a JSON field?

Reply:
> I don't think, however, that the payload needs to be a string for it to be used as a search term in the Notification query

In general JSONB does support equality. The problem here is that the payload received via `LISTEN` is always a string and in python you have a string. I need some experimenting to figure out how to do the search here when the input is a string. 

> I thought the payload was being serialized as JSON via the to_json

String like `"abc"` is a valid json but it just represents the string not the object (check the grammar https://www.json.org/json-en.html) and `to_json` converts a string to a JSON which value is a string. But you want to store a json object (object as in the grammar e.g. `{'a': 1}`). Compare this:

```
# select to_json('{"a": 1}'::text) AS "incorrect", '{"a": 1}'::jsonb AS "correct", '{"a": 1}'::jsonb ->> 'a' AS "access";
  incorrect   | correct  | access
--------------+----------+--------
 "{\"a\": 1}" | {"a": 1} | 1
(1 row)
```

`incorrect` contains a json string and `correct` contains a json object which can be used to get values by key (`access` column)

Reply:
I think this change deserves a separate PR.